modules/persistence/src/services/metadata/repos_branches/index.ts

```diff
@@ -114,20 +115,31 @@ export const getRepoBranchesEntry = async (project: project, branch = ''): Promi
   // get from DB if not cached
   try {
     const db = await pool();
-    const matchCondition = { project };
+    const matchCondition = {
+      project,
+      // We want the repo branches of the single deployable repo for a docset
+      prodDeployable: true,
+    };
     if (branch) {
       matchCondition['branches'] = { $elemMatch: { gitBranchName: branch } };
     }
     const aggregationPipeline = getAggregationPipeline(matchCondition);
 
     const cursor = db.collection('docsets').aggregate(aggregationPipeline);
     const res = (await cursor.toArray()) as unknown as ReposBranchesDocument[];
+    const returnedEntry = res[0];
+
+    if (res.length > 1) {
+      console.warn(
+        `Expected 1 deployable repo for docset with project "${project}", but found ${res.length} instead. Defaulting to first found: "${returnedEntry.repoName}".`
+      );
+    }
 
     // if not already set, set cache value for repo_branches
     if (!internals[project]) {
```

Comment:
@rayangler, this might be a silly question, but I see we are caching the value, but wouldn't we want to check the cache before querying the `docsets`?

Reply:
The cache is checked earlier [here](https://github.com/mongodb/docs-worker-pool/blob/b0253fb8c4c9e9c9437f0b23f6acda2197a29196/modules/persistence/src/services/metadata/repos_branches/index.ts#L105).